scripting/include/messages.inc

```diff
@@ -56,31 +60,159 @@ native int GetUserMsgId(const char[] msg_name);
+ *
+ * @param msg_type          Message index.
+ * @param type              Block type.
+ *
+ * @noreturn
+ */
+native void SetMsgBlock(int msg_type, BlockType type);
+
+/**
+ * @brief Gets number of arguments that were passed to a message.
+ *
+ * @return                  Number of arguments.
+ */
+native int GetMsgArgs();
+
+/**
+ * @brief  Gets the argument type.
+ *
+ * @param argn              Argument number.
+ *
+ * @return                  Argument type.
+ */
+native MsgArgType GetMsgArgType(int argn);
+
+/**
+ * @brief  Gets the integer value of a argument.
+ *
+ * @param argn              Argument number.
+ *
+ * @return                  Argument type.
+ */
+native int GetMsgArgInt(int argn);
+
+/**
+ * @brief  Gets the float value of a argument.
+ *
+ * @param argn              Argument number.
+ *
+ * @return                  Argument type.
+ */
+native float GetMsgArgFloat(int argn);
+
+/**
+ * @brief  Gets the string value of a argument.
+ *
+ * @param argn              Argument number.
+ * @param buffer            Buffer for string.
+ * @param size              Buffer size.
+ *
+ * @return                  String length.
+ */
+native int GetMsgArgString(int argn, char[] buffer, int size);
+
+/**
+ * @brief  Sets the integer value of a argument.
+ *
+ * @param argn              Argument number.
+ * @param value             Argument value.
+ *
+ * @noreturn
+ */
+native void SetMsgArgInt(int argn, int value);
+
+/**
+ * @brief  Sets the float value of a argument.
+ *
+ * @param argn              Argument number.
+ * @param value             Argument value.
  *
  * @noreturn
- * @error              Invalid message index, invalid number of parameters passed.
  */
-native void MessageBegin(int dest, int msgid, int origin[3] = {0,0,0}, int player = 0);
+native void SetMsgArgFloat(int argn, float value);
+
+/**
+ * @brief  Sets the string value of a argument.
+ *
+ * @param argn              Argument number.
+ * @param string            Argument string.
+ *
+ * @noreturn
+ */
+native void SetMsgArgString(int argn, const char[] string);
 
 /**
  * @brief Starts a message.
  *
- * @param dest         Destination type.
- * @param msgid        Message index.
- * @param origin       Message origin.
- * @param player       Player index to broadcast to.
+ * @param dest          Destination type.
+ * @param msgid         Message index.
+ * @param origin        Message origin.
+ * @param player        Player index to broadcast to.
+ * @param ignore_hooks  Ignore this message in message hooks.
  *
  * @noreturn
- * @error              Invalid message index, invalid number of parameters passed.
+ * @error               Invalid message index, invalid number of parameters passed.
  */
-native void MessageBeginF(int dest, int msgid, float origin[3] = {0.0,0.0,0.0}, int player = 0);
+native void MessageBegin(MessageDest dest, int msgid, float origin[3] = {0.0,0.0,0.0}, int player = 0, bool ignore_hooks = true);
```

Comment:
Maybe it would be better to remove "negativeness" from `ignore_hooks`? Something like `bool hookable = false` or `bool catchable = false`.